### PR TITLE
vimc-4136 Add get run status for polling running reports

### PR DESCRIPTION
**Pull request by EmmaLRussell**
Also added unit tests for the api class (only had integration tests before), and a custom error class for response errors from OrderlyWeb.

### CODE REVIEW

File: orderlyweb_api/orderly_web_api.py [hunk 1]
```diff
@@ -9,21 +11,40 @@ def __init__(self, base_url, token):
         response = requests.post(auth_url, headers=headers)
         if response.status_code != 200:
             msg = 'Unexpected status code: {}. Unable to authenticate.'
-            raise Exception(msg.format(response.status_code))
+            raise OrderlyWebResponseError(msg.format(response.status_code),
```
Comment (hillalex) [inline orderlyweb_api/orderly_web_api.py]: Might it be more useful to just pass through the error message from orderly web? These should be readable, informative things like "Invalid token" or "Missing required permissions: ..."

File: orderlyweb_api/orderly_web_api.py [hunk 2]
```diff
@@ -9,21 +11,40 @@ def __init__(self, base_url, token):
...
+    def headers(self):
+        return {'Authorization': 'Bearer {}'.format(self.token)}
 
     def build_url(self, route):
         return '{}/api/v1/{}/'.format(self.base_url, route)
+
+    @staticmethod
+    def handle_response(response):
+        if response.status_code != 200:
+            msg = 'Unexpected status code: {}'
```
Comment (hillalex) [inline orderlyweb_api/orderly_web_api.py]: Similarly here, why obscure the original error message?

File: orderlyweb_api/orderly_web_api.py [hunk 3]
```diff
@@ -9,21 +11,40 @@ def __init__(self, base_url, token):
         response = requests.post(auth_url, headers=headers)
         if response.status_code != 200:
             msg = 'Unexpected status code: {}. Unable to authenticate.'
-            raise Exception(msg.format(response.status_code))
+            raise OrderlyWebResponseError(msg.format(response.status_code),
+                                          response=response)
         self.token = response.json()['access_token']
```
Comment (hillalex) [inline orderlyweb_api/orderly_web_api.py]: So `self.token` is an OW access token, and the `token` that is passed to `__init__` is a Montagu token? Perhaps relname one or both to make this super clear.

**Review by hillalex — COMMENTED** [review]
Looks good to me, couple of small suggestions.

**Comment by EmmaLRussell** [comment]
I've updated the error class to extract error message from the orderly web response, and I changed the scripts to use orderyweb deploy while I was in there.

File: orderlyweb_api/orderly_web_api.py [hunk 4]
```diff
@@ -4,16 +4,14 @@
 
 
 class OrderlyWebAPI:
-    def __init__(self, base_url, token):
+    def __init__(self, base_url, auth_token):
```
Comment (hillalex) [inline orderlyweb_api/orderly_web_api.py]: Sorry to be a pain, but I still don't find `auth_token` and `access_token` to be transparently different terms - can this one be `montagu_token`?

Comment (EmmaLRussell) [reply orderlyweb_api/orderly_web_api.py]: It could theoretically be a Github token though, couldn't it?

Comment (EmmaLRussell) [reply orderlyweb_api/orderly_web_api.py]: Would something like `auth_provider_token` be better?

Comment (hillalex) [reply orderlyweb_api/orderly_web_api.py]: Ah of course, sorry I completely forgot about that! `auth_provider_token` would be perfect

**Review by hillalex — APPROVED** [review]

**Pull request merged**